src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1362 @@
+
+        hidden_states = []
+
+        if return_all_hiddens:
+            hidden_states.append(hidden_state)
+
+        relative_pos_bias = None
+        if self.relative_position is not None:
+            relative_pos_bias = self.relative_position(
+                batch_size=hidden_state.size(0), qlen=hidden_state.size(1), klen=hidden_state.size(1)
+            )
+
+        # past_key_value is not None during inference if we use the bidirectional encoder as a generator as in s2s-ft (https://arxiv.org/abs/2110.13640)
+        for idx, layer in enumerate(self.layers):
+            hidden_state = layer(
+                hidden_state,
+                encoder_padding_mask=encoder_padding_mask if past_key_value is None else None,
+                attention_mask=attention_mask,
+                relative_pos=relative_pos_bias,
+                multiway_split_position=multiway_split_position,
+                past_key_value=past_key_value[idx] if past_key_value is not None else None,
+            )
+            if return_all_hiddens:
+                hidden_states.append(hidden_state)
+
+        if self.fc_norm is not None:
+            hidden_state = self.fc_norm(hidden_state, split_position=multiway_split_position)
+
+        if not return_dict:
+            return [hidden_state, encoder_embedding, hidden_states]
+
+        return Beit3ModelOutput(
+            encoder_out=hidden_state,
+            encoder_embedding=encoder_embedding,
+            hidden_states=hidden_states,
+        )
+
+
+@add_start_docstrings(
+    """BEiT-3 is a general-purpose multimodal foundation model that excels in both vision and vision-language tasks. It
+        utilizes [Multiway transformers] (https://arxiv.org/abs/2208.10442) for deep fusion and modality-specific
+        encoding, and unifies masked modeling on images, texts, and image-text pairs, achieving top performance on
+        multiple benchmarks.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3Model(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.text_embedding = nn.Embedding(config.vocab_size, config.embed_dim)
+        self.vision_embedding = Beit3VisionEmbedding(config)
+        Beit3Embedder(config)
+        self.encoder = Beit3Encoder(config)
+        self.gradient_checkpointing = False
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.text_embedding
+
+    def set_input_embeddings(self, value: nn.Embedding) -> None:
+        self.text_embedding = value
+
+    def get_num_layers(self):
+        return self.encoder.num_layers
+
+    @add_start_docstrings_to_model_forward(BEIT3_MODEL)
+    def forward(
+        self,
+        input_ids=None,
+        pixel_values=None,
+        text_padding_mask=None,
+        attention_mask=None,
+        vision_masked_position=None,
+        past_key_value=None,
+        text_end_positions=None,
+        return_dict=None,
+        output_hidden_states=True,
```

Comment:
```output_hidden_states``` is unused in this method.
I found that this causes some issues when fine tuning using Beit3ImageClassification.
During the evaluation, gpu memory usage grows massively. The same code works with Beit1.
Looks like the evaluation_loop in trainer uses hidden state instead of logits because ```output_hidden_states``` flag is ignored here.
L